Project Code from Steve Xu

DongkunXu opened the pull request:
This is the version of the project from Steve Xu (s2523105), for MSc course "Embedded Mobile and Wireless Systems (EWireless)".

fzampella-huawei commented:
Please specify the issues to address in this pull request, not just add all the changes, ideally we will add each issue (or a few if needed) as a pull request so we can improve the code step by step.

You can add the issues in the "development" setting of the pull request

DongkunXu commented:
Thank you for your response.

I apologize for any confusion. I am not entirely sure if you are requesting me to identify the issues currently present in my code or to list the changes I have made. 

However, if you are looking for a list of the changes I made in the project, here are the main updates in the current code:

1. Added WiFi upload code for server interaction, including checks and decoding of return values.
2. Integrated interactions with barometer, GNSS, angle, and other sensors to display and update markers on the map.
3. Added multi-floor indoor maps for Nucleus building and Noreen and Kenneth Murray Library, with support for automatic display and hiding based on user location, and automatic floor switching based on user altitude.
4. Implemented a dynamic batch optimization algorithm to real-time integrate WiFi position and PDR data for optimal positioning information, and to plot user trajectories on the map.
5. Removed the original recording page; all information is now displayed directly on the map. The UI has been optimized to accommodate the new features, including auto-showing floor buttons and a series of functional buttons and markers to enhance user interaction with the map.

Please let me know if you need further information.

fzampella-huawei commented:
I am interested in specific improvements not the whole set of changes, for example could you separate the changes to handle the wifi positioning through openpositioning (leave the display as another request for now, as it has too many changes). I know today is the deadline, but given that you showed interest in contributing I will give you a couple of days to finish it and to take it into account for the awards ceremony

From: DongkunXu ***@***.***>
Sent: 15 May 2024 16:19
To: openpositioning/PositionMe ***@***.***>
Cc: Francisco Zampella ***@***.***>; Review requested ***@***.***>
Subject: Re: [openpositioning/PositionMe] Project Code from Steve Xu (PR #31)


Thank you for your response.

I apologize for any confusion. I am not entirely sure if you are requesting me to identify the issues currently present in my code or to list the changes I have made.

However, if you are looking for a list of the changes I made in the project, here are the main updates in the current code:

  1.  Added WiFi upload code for server interaction, including checks and decoding of return values.
  2.  Integrated interactions with barometer, GNSS, angle, and other sensors to display and update markers on the map.
  3.  Added multi-floor indoor maps for Nucleus building and Noreen and Kenneth Murray Library, with support for automatic display and hiding based on user location, and automatic floor switching based on user altitude.
  4.  Implemented a dynamic batch optimization algorithm to real-time integrate WiFi position and PDR data for optimal positioning information, and to plot user trajectories on the map.
  5.  Removed the original recording page; all information is now displayed directly on the map. The UI has been optimized to accommodate the new features, including auto-showing floor buttons and a series of functional buttons and markers to enhance user interaction with the map.

Please let me know if you need further information.

—
Reply to this email directly, view it on GitHub<https://github.com/openpositioning/PositionMe/pull/31#issuecomment-2112838843>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A6J27J3EMEPFLSRZTN3FBXTZCN4HLAVCNFSM6AAAAABHYGCIJOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMJSHAZTQOBUGM>.
You are receiving this because your review was requested.Message ID: ***@***.***>

DongkunXu commented:
Thank you very much for your response and consideration!

I understand. I will sort out the specific improvements I have made and submit them accordingly as soon as possible.

DongkunXu commented:
Additionally, I would like to clarify whether each pull request should only contain the respective changes. I have made some UI and other changes, and submitting partial code changes might lead to some features not being functional or the app not being fully operational. These changes might only serve as demonstrations of code and functionality improvements rather than a complete, working application. 

Is it acceptable to submit each pull request with the complete, final version of the code but highlight the specific changes and their locations in my commit?

fzampella-huawei commented:
It should only contain the isolated changes, (no UI changes unless you are implementing an issue regarding the UI or if it is a requirement, explain in the merge request), that way it opens the opportunity to collaborate and add changes from different teams

DongkunXu commented:
Got it, thank you!
